File: Appli/Méthodologie/Etapes/Analyse/Benchmarking/Code sources/bench_Java_c/benchMap/src/com/esiag/isidis/ListeProduit.java
package com.esiag.isidis;

import java.util.HashMap;
import java.util.Hashtable;
import java.util.Map;

public class ListeProduit {

	private Map<String, Produit> liste;

	public ListeProduit(Map<String, Produit> liste) {
		super();
		this.liste = liste;
	}

	public ListeProduit() {
		super();
		liste=new HashMap<>();
	}
	public void insert(Produit p)
	{
		liste.put(p.getRef(), p);
		
	}
	public Produit select(String ref)
	{
		return liste.get(ref);
	}
	
	
}

File: Appli/Méthodologie/Etapes/Analyse/Benchmarking/Code sources/bench_Java_c/WindowsFormsApplication1/WindowsFormsApplication1/ListeProduit.cs
using System;
using System.Collections;
using System.Linq;
using System.Text;

namespace WindowsFormsApplication1
{
    class ListeProduit
    {
        private Hashtable tab;

        public ListeProduit()
        {
            tab = new Hashtable();
            
        }
        public void insert(Produit p)
        {
            tab.Add(p.Reference,p);
        }
         public Produit select(String r)
        {
            return (Produit)tab[r];
        }
    }
}
